### PR TITLE
Add Python 3.10 support

### PR DESCRIPTION
**Pull request by thedrow**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

## Description

We will support Python 3.10 starting from Celery 5.2.
We should start testing with it now to figure out what kind of problems we may have.

### CODE REVIEW

**Review by auvipy — APPROVED** [review]
looks good, merge when 3.10 stable released

**Comment by thedrow** [comment]
> 
> 
> looks good, merge when 3.10 stable released

This won't fail the build anyway starting from 7ff872b.
Is there any problem merging this just so we'll know what's the status of our Python 3.10 support?

**Review by maybe-sybr — APPROVED** [review]
lgtm, I reckon we merge and make a follow up issue to bump along the release cycle defined in PEP-0619
> https://www.python.org/dev/peps/pep-0619/

Keep in mind, @thedrow, that the release schedule for Python 3.10 will define our release schedule for 5.2 if we want to release with official support. Given our vaguely accelerating release cycle, I'd say we could alter that expectation slightly to keep the door open for a celery 5.2 before October with wording in the changelog about provisional 3.10 support which we could turn official in a 5.2.x release when the Python 3.10 rolls around. Depends on your views on the 5.2 release cycle timeline and what we want to land in it.

If @auvipy doesn't merge this in the next 8ish hours, I'll do it at the end of my work day.

File: tox.ini [hunk 1]
```diff
@@ -1,7 +1,9 @@
 [tox]
+requires =
+    tox-gh-actions
```
Comment (maybe-sybr) [inline tox.ini:3]: Do we need this here since we install it in the actions yaml? No big deal, we can merge without an answer on this, it should be fine

Comment (thedrow) [reply tox.ini:3]: It's seems like a good idea to specify the deps for tox to work.

Comment (graingert) [reply tox.ini:3]: it's best to specify tox requires here

Comment (maybe-sybr) [reply tox.ini:3]: :+1:

**Pull request merged**